### PR TITLE
update wx_shape validator optional syntax

### LINKED ISSUE #184: redefine "optional" syntax for wx_shape validation
thinking about #176:

the current implementation (from #176) allows shape definition for optional properties (good), however I see the following problem:

- the optional indication like `(optional_prop)` is redundant with the `required` field (and must be kept in sync). Every field that is not a required property has to have the shape attribute as optional for the schema to make sense.

### proposal:

1. only use ASDF default `required: ...` to indicate required fields
2. allow shape validation to pass **by default** if a property is missing (since this schema violation should be caught by `required`)
3. rework the `required_shape / (optional_shape)` syntax to do the following:
    - if the property exists and a shape is *required*, the property **must** report a shape **and** validate against the shape requirements
    - if the property exists and a shape is *optional*, the property **must** validate against the shape requirements **only if** it reports a shape
    - if the property exists and a shape is *optional*, the property passes validation if it does not report a shape

### Example 1:
```yaml
properties:
  required_prop:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
  optional_prop:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
wx_shape:
  required_prop: [n]
  optional_prop: [n]
required: [required_prop]
```
This is the default case for optional properties. If `optional_prop` exists, it must validate correctly. If it does not exist it is skipped during `wx_shape` validation.

### Example 2:
```yaml
properties:
  required_prop:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
  optional_prop:
    oneOf:
      - tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
      - tag: **/timedeltaindex-1.0.0
      - type: string
wx_shape:
  required_prop: [n]
  (optional_prop): [n]
required: [required_prop]
```
This is an example that shows the new behavior.
- If `optional_prop` does not exist, it is simply skipped (like example 1)
- If `optional_prop` does exist and is a `ndarray`, it must validate the `[n]` shape (because we can evaluate the shape of `ndarray`)
- If `optional_prop` does exist and is a `timedeltaindex`, it must validate the `[n]` shape (because we can evaluate the shape of `timedeltaindex` from our custom code)
- If `optional_prop` does exist and is a `string`, shape validation is skipped (because we don't define a shape property of `string`)

### Example 3:
```yaml
properties:
  required_prop:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
  optional_prop:
    tag: tag:stsci.edu:asdf/core/ndarray-1.0.0
wx_shape:
  required_prop: [n]
  (optional_prop): [n]
required: [required_prop]
```
This case shows the same behavior as Example 1, and is redundant since all possibilities of `optional_prop` define a shape. It should be avoided.

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
see #184

## Related Issues
Closes #184

## Checks

- [x] updated CHANGELOG.rst
- [x] updated tests
- [x] updated doc/
- [ ] update example/tutorial notebooks
- [ ] update manifest file

### CODE REVIEW

**Comment by CagtayFabry** [comment]
I have added another way to indicate a optional shape requirement by enclosing the list in braces for instances where `wx_shape` is defined directly at the property
```yaml
wx_shape: ([1,2,a])
```

File: weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml [hunk 1]
```diff
@@ -103,20 +103,18 @@ properties:
     tag: "asdf://weldx.bam.de/weldx/tags/time/timedeltaindex-0.1.*"
   coordinates:
     description: |
-      An ndarray containing the coordinates.
+      A ndarray containing the coordinates.
```
Comment (vhirtham) [inline weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml:106]: I think the previous version was correct due to the way `ndarray` is pronounced. But not 100% sure here

File: weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml [hunk 2]
```diff
@@ -103,20 +103,18 @@ properties:
...
 
     oneOf:
       - tag: "asdf://weldx.bam.de/weldx/tags/core/variable-0.1.*"
       - tag: "asdf://weldx.bam.de/weldx/tags/core/time_series-0.1.*"
   orientations:
     description: |
-      An ndarray containing the orientations.
+      A ndarray containing the orientations.
```
Comment (vhirtham) [inline weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml:113]: Same as above

**Review by vhirtham — APPROVED** [review]

File: weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml (hunk 2 above)
Comment (CagtayFabry) [reply weldx/schemas/weldx.bam.de/weldx/core/transformations/local_coordinate_system-0.1.0.yaml:113]: true! I just auto-accepted the PyCharm spelling correction

**Pull request merged**